rotkehlchen/api/v1/encoding.py

```diff
@@ -114,7 +115,11 @@ def _deserialize(
     ) -> Timestamp:
         try:
             timestamp = deserialize_timestamp(value)
-        except DeserializationError as e:
+            # NB: do not include the statement below in 'deserialize_timestamp()'
+            # as the backend deserializes timestamps in milliseconds across the
+            # app (e.g. exchanges) and it would raise an out of range ValueError
+            datetime.utcfromtimestamp(timestamp)
```

Comment:
Looking at the code I've realised that it wouldn't be safe to validate a timestamp (as a valid date) in `deserialize_timestamp` as some logic (e.g. exchanges) pass to it timestamps in milliseconds.
I think we could consider extending `deserialize_timestamp` args with a kwarg `in_milliseconds` (defaulted to False). If so, divide by 1000 and validate it is a valid timestamp. Then refactor any related code for using it.

Reply:
This is a timestamp given by the API. Not by exchanges. We only accept second timestamps there. This is our own input, not someone else's.